Automattic/wp-calypso
# composite-checkout: Add WeChat Pay

#### Changes proposed in this Pull Request

This adds the "WeChat Pay" payment method to composite-checkout.

#### Screenshots

Old checkout:

![Screen Shot 2020-06-29 at 12 53 18 PM](https://user-images.githubusercontent.com/2036909/86034904-70e8d300-ba09-11ea-9d8a-1ba89935a068.png)

![wechat-old-qr](https://user-images.githubusercontent.com/2036909/86188998-50ef0780-bb0e-11ea-8e7f-0ce0f9c8772c.png)


New checkout:

![wechat-new-open](https://user-images.githubusercontent.com/2036909/86189000-53e9f800-bb0e-11ea-8528-10f976058615.png)

![wechat-new-qr](https://user-images.githubusercontent.com/2036909/86189005-56e4e880-bb0e-11ea-9ad5-591f196c7cf8.png)

#### Testing instructions

- Apply D45650-code to force WeChat Pay to be available.
- Sandbox the store.
- Add a plan to your cart and visit checkout. 
- Add `?flags=composite-checkout-force` to the query string in the URL and reload the page because currently it is disabled for international currencies.
- Verify that "WeChat Pay" is an option displayed in the final payment step.
- Verify that you cannot pay if the "Name" field is empty.
- Fill out the "Name" field and submit the payment.
- Verify that you see a QR code appear and it is scrolled into view.
- In a *new window or tab* open the link at the bottom that reads "To open and pay with the wechat pay app directly, click here".
- In the new window, you should see a Stripe test page.
- Click to accept the payment and verify that the payment completes successfully in the original window and that the plan is purchased.
